06.bowling_object/game.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def initialize(scores)
+    i = 0
+    @frames = []
+    (1..10).each do |frame|
+      if frame == 10
+        @frames << Frame.new(scores[i], scores[i + 1], scores[i + 2])
+      elsif scores[i] == 'X'
+        @frames << Frame.new('X', 0)
+        i += 1
+      else
+        @frames << Frame.new(scores[i], scores[i + 1])
+        i += 2
+      end
+    end
+  end
+
+  def calc_result
+    result = 0
+    @frames.each.with_index do |frame, i|
+      result += frame.frame_score
+      if i < 9
+        if frame.strike?
+          result += strike_score_bonus(i + 1)
+        elsif frame.spare?
+          result += spare_score_bonus(i + 1)
+        end
+      end
+    end
+    result
```

Comment:
```suggestion
    @frames.sum do |frame|
      frame.calc_score(@frames)
    end
```

Gameが点数の計算のロジックを持ちすぎているので、それよりもFrameクラスにロジックを寄せた方がオブジェクト指向らしいかも。

たとえば、frameに全フレームのデータを渡せば、自分で次のフレームや次の次のフレームのデータを使って点数計算ができます。
また、Frame.newするタイミングでフレーム番号を渡せば、自分が何フレーム目なのかもframe自身で取得できます。